### PR TITLE
doc: how to set version number in build

### PR DESCRIPTION
**Pull request by pjfanning**
* see https://github.com/apache/incubator-pekko/issues/339

### CODE REVIEW

**Comment by mdedetrich** [comment]
Is there a reason why you don't want to document doing https://github.com/apache/incubator-pekko/issues/339#issuecomment-1561848206 ? Its far easier than having to create a file (which makes the source dirty)

**Comment by pjfanning** [comment]
> Is there a reason why you don't want to document doing [#339 (comment)](https://github.com/apache/incubator-pekko/issues/339#issuecomment-1561848206) ? Its far easier than having to create a file (which makes the source dirty)

* these notes are really aimed at the Incubator PMC voters - the majority of whom are not Scala or sbt users
* the source release when unzipped is not under git control
* adding the file once means that you don't need to set it the prop in every sbt call

**Comment by mdedetrich** [comment]
Would it be a big deal to document both approaches (you can mention the easier approach at the start)? Alternately I can also document the second approach myself after this PR is merged.

**Comment by pjfanning** [comment]
@mdedetrich I made some changes. I'll leave this open for a few days to facilitate discussion.

**Review by mdedetrich — APPROVED** [review]
lgtm

File: README.md [hunk 1]
```diff
@@ -39,6 +39,12 @@ See https://pekko.apache.org for the documentation including the API docs. The d
      - the `index.html` file will appear in `target/paradox/site/main/`
 - `sbt unidoc` will build the Javadocs for all the modules and load them to one place (may require Graphviz, see Prerequisites above)
      - the `index.html` file will appear in `target/scala-2.13/unidoc/`
+- `sbt sourceDistGenerate` will generate source release to `target/dist/`
+- The version number that appears in filenames and docs is derived, by default. The derived version contains the most git commit id or the date/time (if the directory is not under got control).
```
Comment (mdedetrich) [inline README.md:43]: type here, `under got control` should be `under git control`

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Changing to request changes because I found a typo

**Review by mdedetrich — APPROVED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (pjfanning) [reply README.md]: thanks - fixed the typo

**Review by He-Pin — APPROVED** [review]
lgtm

**Pull request merged**